activity_heartbeating/my_activities.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'temporalio/activity'
+
+module ActivityHeartbeating
+  module MyActivities
+    # Activity that demonstrates progress tracking with heartbeating and cancellation
+    class FakeProgress < Temporalio::Activity::Definition
+      def execute(sleep_interval_ms = 1000)
+        context = Temporalio::Activity::Context.current
+
+        begin
+          # Allow for resuming from heartbeat details if available
+          starting_point = context.info.heartbeat_details.first || 1
+
+          context.logger.info("Starting activity at progress: #{starting_point}")
+
+          (starting_point..100).each do |progress|
+            # Sleep for the interval - checking cancellation after sleep
+            sleep(sleep_interval_ms / 1000.0)
+
+            # Check if activity was canceled
+            if context.cancellation.canceled?
+              context.logger.info("Fake progress activity cancelled at progress: #{progress}")
+              raise Temporalio::Error::CanceledError, "Activity was canceled at progress: #{progress}"
+            end
```

Comment:
Is this block of code needed? I would expect cancellation to raise so is there any reason to also explicitly check for cancel?

Reply:
You're correct, it works fine without this.